smartsim/entity/strategies.py

```diff
@@ -90,32 +95,69 @@ def _impl(
 # single application if parameters only have one value
 @_register("all_perm")
 def create_all_permutations(
-    params: t.Mapping[str, t.Sequence[str]],
+    file_params: t.Mapping[str, t.Sequence[str]],
+    exe_arg_params: t.Mapping[str, t.Sequence[t.Sequence[str]]],
     _n_permutations: int = 0,
```

Comment:
nit: now that this parameter is used, feel free to remove the leading underscore
```suggestion
    n_permutations: int = 0,
```